src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs

```diff
@@ -172,5 +174,55 @@ public void ProcFs_NotEmpty(string path)
         {
             Assert.InRange(File.ReadAllBytes(path).Length, 1, int.MaxValue);
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)] // DOS device paths (\\.\ and \\?\) are a Windows concept
+        public async Task ReadAllBytes_NonSeekableFileStream_InWindows()
+        {
+            string pipeName = FileSystemTest.GetNamedPipeServerStreamName();
+            string pipePath = Path.GetFullPath($@"\\.\pipe\{pipeName}");
+
+            var namedPipeWriterStream = new NamedPipeServerStream(pipeName, PipeDirection.Out);
+            var contentBytes = new byte[] { 1, 2, 3 };
+
+            using (var cts = new CancellationTokenSource())
+            {
+                Task writingServerTask = WaitConnectionAndWritePipeStreamAsync(namedPipeWriterStream, contentBytes, cts.Token);
+                Task<byte[]> readTask = Task.Run(() => File.ReadAllBytes(pipePath), cts.Token);
+                cts.CancelAfter(TimeSpan.FromSeconds(10));
+
+                await writingServerTask;
+                byte[] readBytes = await readTask;
+                Assert.Equal<byte>(contentBytes, readBytes);
+            }
+
+            static async Task WaitConnectionAndWritePipeStreamAsync(NamedPipeServerStream namedPipeWriterStream, byte[] contentBytes, CancellationToken cancellationToken)
+            {
+                await using (namedPipeWriterStream)
+                {
+                    await namedPipeWriterStream.WaitForConnectionAsync(cancellationToken);
+                    await namedPipeWriterStream.WriteAsync(contentBytes, cancellationToken);
+                }
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public async Task ReadAllBytes_NonSeekableFileStream_InLinux()
+        {
+            var path = "/dev/tty";
```

Comment:
Looks like in some helix environments, `/dev/tty` is not mounted in docker container. We can skip the test in that case:
```suggestion
            var path = "/dev/tty";
            if (!Path.Exists(path))
            {
                throw new SkipTestException(path + " is not available in this environment.");
            }
```

Reply:
This tests could be written using some other files that don't depend on a terminal.
For example using `RemoteExecutor`, redirect streams, and use `/dev/std{in,out,err}`.

Reply:
Yup, though those three may get special treatment by the way of ENXIO handling: https://github.com/dotnet/corefx/pull/3352.

`/dev/zero`, `/dev/null` being nop, we are left with `/dev/tty` and `/dev/console` which can give us pretty decent coverage for this case (as it is easier to read back and assert). With .NET 5, Write works just fine and this test case throws on ReadAllBytes because stream is unseekable (the very aspect this PR is fixing). But with .NET 6 and main, right now Write is failing which https://github.com/dotnet/runtime/pull/59846 will fix.

Still worth adding additional cases for `/dev/std{in,out,err}`, IMO.